scrapy/http/request/form.py

```diff
@@ -107,8 +108,13 @@ def _get_inputs(form, formdata, dont_click, clickdata, response):
 
     inputs = form.xpath('descendant::textarea'
                         '|descendant::select'
-                        '|descendant::input[@type!="submit" and @type!="image" and @type!="reset"'
-                        'and ((@type!="checkbox" and @type!="radio") or @checked)]')
+                        '|descendant::input[not(@type) or @type['
+                        ' translate(., "SUBMIT", "submit") != "submit"'
```

Comment:
lxml has support for EXSLT regex extensions:
https://stackoverflow.com/questions/2755950/how-to-use-regular-expression-in-lxml-xpath
http://exslt.org/regexp/functions/test/index.html
```
boolean regexp:test(string, string, string?)
```

I think it would be easier to read